src/rdkafka_partition.c

```diff
@@ -4426,6 +4426,91 @@ int rd_kafka_topic_partition_list_regex_cnt(
 }
 
 
+/**
+ * @brief Match function that returns true if topic is not a regex.
+ */
+static int rd_kafka_topic_partition_not_regex(const void *elem,
+                                              const void *opaque) {
+        const rd_kafka_topic_partition_t *rktpar = elem;
+        return *rktpar->topic != '^';
+}
+
+/**
+ * @brief Return a new list with all regex topics removed.
+ *
+ * @remark The caller is responsible for freeing the returned list.
+ */
+rd_kafka_topic_partition_list_t *rd_kafka_topic_partition_list_remove_regexes(
+    const rd_kafka_topic_partition_list_t *rktparlist) {
+        rd_kafka_topic_partition_list_t *non_regex_rktparlist =
```

Comment:
This variable can be removed and it can directly return the `rd_kafka_topic_partition_list_match` returned value